File: hailiao-api/src/main/java/com/hailiao/api/dto/UserSessionDTO.java
package com.hailiao.api.dto;

import java.util.Date;

public class UserSessionDTO {
    private String sessionId;
    private String deviceId;
    private String deviceName;
    private String deviceType;
    private String loginIp;
    private Boolean active;
    private Boolean currentSession;
    private Date createdAt;
    private Date lastActiveAt;
    private Date revokedAt;

    public String getSessionId() { return sessionId; }
    public void setSessionId(String sessionId) { this.sessionId = sessionId; }

    public String getDeviceId() { return deviceId; }
    public void setDeviceId(String deviceId) { this.deviceId = deviceId; }

    public String getDeviceName() { return deviceName; }
    public void setDeviceName(String deviceName) { this.deviceName = deviceName; }

    public String getDeviceType() { return deviceType; }
    public void setDeviceType(String deviceType) { this.deviceType = deviceType; }

    public String getLoginIp() { return loginIp; }
    public void setLoginIp(String loginIp) { this.loginIp = loginIp; }

    public Boolean getActive() { return active; }
    public void setActive(Boolean active) { this.active = active; }

    public Boolean getCurrentSession() { return currentSession; }
    public void setCurrentSession(Boolean currentSession) { this.currentSession = currentSession; }

    public Date getCreatedAt() { return createdAt; }
    public void setCreatedAt(Date createdAt) { this.createdAt = createdAt; }

    public Date getLastActiveAt() { return lastActiveAt; }
    public void setLastActiveAt(Date lastActiveAt) { this.lastActiveAt = lastActiveAt; }

    public Date getRevokedAt() { return revokedAt; }
    public void setRevokedAt(Date revokedAt) { this.revokedAt = revokedAt; }
}

File: hailiao_flutter/lib/models/user_session_dto.dart
import 'package:json_annotation/json_annotation.dart';

part 'user_session_dto.g.dart';

@JsonSerializable()
class UserSessionDTO {
  @JsonKey(name: 'sessionId')
  final String? sessionId;

  @JsonKey(name: 'deviceId')
  final String? deviceId;

  @JsonKey(name: 'deviceName')
  final String? deviceName;

  @JsonKey(name: 'deviceType')
  final String? deviceType;

  @JsonKey(name: 'loginIp')
  final String? loginIp;

  @JsonKey(name: 'active')
  final bool? active;

  @JsonKey(name: 'currentSession')
  final bool? currentSession;

  @JsonKey(name: 'createdAt')
  final String? createdAt;

  @JsonKey(name: 'lastActiveAt')
  final String? lastActiveAt;

  @JsonKey(name: 'revokedAt')
  final String? revokedAt;

  UserSessionDTO({
    this.sessionId,
    this.deviceId,
    this.deviceName,
    this.deviceType,
    this.loginIp,
    this.active,
    this.currentSession,
    this.createdAt,
    this.lastActiveAt,
    this.revokedAt,
  });

  factory UserSessionDTO.fromJson(Map<String, dynamic> json) =>
      _$UserSessionDTOFromJson(json);

  Map<String, dynamic> toJson() => _$UserSessionDTOToJson(this);
}
